pkg/app/api/grpcapi/web_api.go

```diff
@@ -1312,190 +1316,30 @@ func (a *WebAPI) GetInsightData(ctx context.Context, req *webservice.GetInsightD
-}
-
-// getInsightDataForDeployFrequency accumulate insight data in target range for deploy frequency.
-// This function is temporary implementation for front end.
-func (a *WebAPI) getInsightDataForDeployFrequency(
-	ctx context.Context,
-	projectID string,
-	applicationID string,
-	targetRangeFrom time.Time,
-	targetRangeTo time.Time) (*model.InsightDataPoint, error) {
-	filters := []datastore.ListFilter{
-		{
-			Field:    "ProjectId",
-			Operator: "==",
-			Value:    projectID,
-		},
-		{
-			Field:    "CreatedAt",
-			Operator: ">=",
-			Value:    targetRangeFrom.Unix(),
-		},
-		{
-			Field:    "CreatedAt",
-			Operator: "<",
-			Value:    targetRangeTo.Unix(), // target's finish time on unix time
-		},
-	}
-
-	if applicationID != "" {
-		filters = append(filters, datastore.ListFilter{
-			Field:    "ApplicationId",
-			Operator: "==",
-			Value:    applicationID,
-		})
-	}
-
-	pageSize := 50
-	deployments, err := a.deploymentStore.ListDeployments(ctx, datastore.ListOptions{
-		PageSize: pageSize,
-		Filters:  filters,
-	})
-	if err != nil {
-		a.logger.Error("failed to get deployments", zap.Error(err))
-		return nil, status.Error(codes.Internal, "Failed to get deployments")
-	}
-
-	return &model.InsightDataPoint{
-		Timestamp: targetRangeFrom.Unix(),
-		Value:     float32(len(deployments)),
-	}, nil
-}
-
-// getInsightDataForChangeFailureRate accumulate insight data in target range for change failure rate
-// This function is temporary implementation for front end.
-func (a *WebAPI) getInsightDataForChangeFailureRate(
-	ctx context.Context,
-	projectID string,
-	applicationID string,
-	targetRangeFrom time.Time,
-	targetRangeTo time.Time) (*model.InsightDataPoint, error) {
+	count := int(req.DataPointCount)
+	from := time.Unix(req.RangeFrom, 0)
 
-	commonFilters := []datastore.ListFilter{
-		{
-			Field:    "ProjectId",
-			Operator: "==",
-			Value:    projectID,
-		},
-		{
-			Field:    "CreatedAt",
-			Operator: ">=",
-			Value:    targetRangeFrom.Unix(),
-		},
-		{
-			Field:    "CreatedAt",
-			Operator: "<",
-			Value:    targetRangeTo.Unix(), // target's finish time on unix time
-		},
-	}
-
-	if applicationID != "" {
-		commonFilters = append(commonFilters, datastore.ListFilter{
-			Field:    "ApplicationId",
-			Operator: "==",
-			Value:    applicationID,
-		})
-	}
-
-	filterForSuccessDeploy := []datastore.ListFilter{
-		{
-			Field:    "Status",
-			Operator: "==",
-			Value:    model.DeploymentStatus_DEPLOYMENT_SUCCESS,
-		},
-	}
-
-	filterForFailureDeploy := []datastore.ListFilter{
-		{
-			Field:    "Status",
-			Operator: "==",
-			Value:    model.DeploymentStatus_DEPLOYMENT_FAILURE,
-		},
-	}
-
-	pageSize := 50
-	successDeployments, err := a.deploymentStore.ListDeployments(ctx, datastore.ListOptions{
-		PageSize: pageSize,
-		Filters:  append(filterForSuccessDeploy, commonFilters...),
-	})
+	chunks, err := a.insightstore.LoadChunks(ctx, claims.Role.ProjectId, req.ApplicationId, req.MetricsKind, req.Step, from, count)
```

Comment:
Can you add a TODO (before this line) about caching the insight data inside the cache service?

Reply:
OK 👍